tests/unit/web/ServerTests.cpp

```diff
@@ -232,22 +238,21 @@ makeServerSync(
 )
 {
     auto server = std::shared_ptr<web::HttpServer<Executor>>();
-
-    std::mutex m;
-    std::condition_variable cv;
-    bool ready = false;
-
-    boost::asio::dispatch(ioc.get_executor(), [&]() mutable {
-        server = web::makeHttpServer(config, ioc, dosGuard, handler, cache);
-        {
-            std::lock_guard const lk(m);
-            ready = true;
+    auto syncData = std::make_shared<SyncData>();
+
+    boost::asio::dispatch(
+        ioc.get_executor(), [&server, &config, &ioc, &dosGuard, &handler, &cache, syncData]() mutable {
+            server = web::makeHttpServer(config, ioc, dosGuard, handler, cache);
```

Comment:
`boost::asio::spawn(..., boost::asio::use_future).wait()` would be more simple.
Also why couldn't we create server synchronously?

Reply:
Actually, no clue why we did not already have it synchronous but you are right - this whole thing can be simplified to one line 👍